recipes/proj/all/conanfile.py

```diff
@@ -131,9 +131,9 @@ def package_info(self):
         cmake_config_filename = "proj" if proj_version >= "7.0.0" else "proj4"
         cmake_namespace = "PROJ" if proj_version >= "7.0.0" else "PROJ4"
         self.cpp_info.set_property("cmake_file_name", cmake_config_filename)
-        self.cpp_info.set_property("cmake_target_name", cmake_namespace)
+        self.cpp_info.set_property("cmake_target_name", "{0}::{0}".format(cmake_namespace))
         self.cpp_info.set_property("pkg_config_name", "proj")
-        self.cpp_info.components["projlib"].set_property("cmake_target_name", "proj")
+        self.cpp_info.components["projlib"].set_property("cmake_target_name", "{}::proj".format(cmake_namespace))
         self.cpp_info.components["projlib"].set_property("pkg_config_name", "proj")
```

Comment:
To remove or not, maybe keep it to avoid side effect in CMakeDeps? You want to keep the component for cmake_find_package compatibility, but you don't want this component in CMakeDeps.

Reply:
I think it should be fine to not define the target name for this one in CMakeDeps as long as it does not collide with other targets declared